### PR TITLE
Add PACT Provider for claim task endpoint

### PR DESCRIPTION
**Pull request by BenNzewi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-449

### Change description ###
Add PACT Provider for claim task endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

**Review by lewis150193 — DISMISSED** [review]

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagerClaimTaskProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+
+@ExtendWith(SpringExtension.class)
+@Provider("wa_task_management_api_claim_task_by_id")
+//Uncomment this and comment the @PacBroker line to test TaskManagerClaimTaskConsumerTest local consumer.
+//@PactFolder("target/pacts")
+@PactBroker(scheme = "${PACT_BROKER_SCHEME:http}",
+    host = "${PACT_BROKER_URL:localhost}", port = "${PACT_BROKER_PORT:9292}", consumerVersionSelectors = {
+    @VersionSelector(tag = "master")})
```
Comment (alexbaeza) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/TaskManagerClaimTaskProviderTest.java:31]: Question: is this needed or can we use the configuration on the build.gradle?

**Review by alexbaeza — DISMISSED** [review]
LGTM but added minor comment

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/service/TaskManagementGetTaskByIdPactTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import uk.gov.hmcts.reform.wataskmanagementapi.services.CamundaService;
+
+import java.time.ZonedDateTime;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(SpringExtension.class)
+@Provider("wa_task_management_api_get_task_by_id")
```
Comment (iguannago) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/service/TaskManagementGetTaskByIdPactTest.java:31]: do we have a consumer test for this provider? I cannot see it.

Comment (BenNzewi) [reply src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/service/TaskManagementGetTaskByIdPactTest.java:31]: No, the plan s to write one. 
ExUI is the consuming team and they will write their own consumer. The local consumer is our own view of things to ensure we can test locally or we have to wait for the consuming team to tell us if the Provider/Consumer broke when they implement their consumer.

**Comment by alexbaeza** [comment]
Closing as it has been superseeded.

**Pull request closed without merging**